assets/js/util/i18n.js

```diff
@@ -33,28 +33,51 @@ import { __, sprintf } from '@wordpress/i18n';
  * For example, passing 65 returns '1m 5s'.
  *
  * @since 1.0.0
+ * @since n.e.x.t Refactored and renamed to improve localization.
+ * @private
  *
- * @param {number} seconds The number of seconds.
- * @return {string} Human readable string indicating time elapsed.
- *
+ * @param {number}                     seconds   The number of seconds.
+ * @param {(Intl.NumberFormatOptions)} [options] Optional formatting options.
+ * @return {string}   		Human readable string indicating time elapsed.
  */
-export const prepareSecondsForDisplay = ( seconds ) => {
+const durationFormat = ( seconds, options = { unitDisplay: 'short' } ) => {
 	seconds = parseInt( seconds, 10 );
 
-	if ( isNaN( seconds ) || 0 === seconds ) {
-		return '0.0s';
+	if ( Number.isNaN( seconds ) ) {
+		seconds = 0;
 	}
-	const results = {};
-	results.hours = Math.floor( seconds / 60 / 60 );
-	results.minutes = Math.floor( ( seconds / 60 ) % 60 );
-	results.seconds = Math.floor( seconds % 60 );
 
-	const returnString =
-		( results.hours ? results.hours + 'h ' : '' ) +
-		( results.minutes ? results.minutes + 'm ' : '' ) +
-		( results.seconds ? results.seconds + 's ' : '' );
+	let hours = Math.floor( seconds / 60 / 60 ) || '';
+	let minutes = Math.floor( ( seconds / 60 ) % 60 ) || '';
+
+	seconds = Math.floor( seconds % 60 );
+
+	if ( hours ) {
+		hours = numberFormat( hours, {
+			style: 'unit',
+			unit: 'hour',
+			...options,
```

Comment:
Let's move where the spread happens since the `style` and `unit` shouldn't be overridable.

Reply:
Good catch